### PR TITLE
Refactor Update Channel

### PR DESCRIPTION
**Pull request by harryurban**
## Changelog

- Refactor channel.updateChannel to receive the updated RawChannel instead of multiple changed fields
- Add tests for modification in updateChannel
- Remove generic RawChannel Reads, because they can't move deprecated data to new fields
- Add Manual RawChannelReads with moving deprecated data
- Add Tests for RawChannelReads and deprecated data

### CODE REVIEW

File: raw/src/main/scala/de/welt/contentapi/raw/models/RawFormats.scala [hunk 1]
```diff
@@ -150,7 +150,34 @@ object RawReads {
       case err@_ ⇒ jsErrorInvalidJson(err)
     }
   }
-  implicit lazy val rawChannelReads: Reads[RawChannel] = Json.reads[RawChannel]
```
Comment (harryurban) [inline raw/src/main/scala/de/welt/contentapi/raw/models/RawFormats.scala]: remove generic Reads
replace with manual-written Reads, that cope with deprecated fields

File: raw/src/test/scala/de/welt/contentapi/raw/models/RawChannelReadsTest.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package de.welt.contentapi.raw.models
+
+import org.scalatestplus.play.PlaySpec
+import play.api.libs.json.{JsArray, JsNumber, JsObject, JsString, Json}
+
+class RawChannelReadsTest extends PlaySpec {
+
+  "RawChannelReads" should {
```
Comment (harryurban) [inline raw/src/test/scala/de/welt/contentapi/raw/models/RawChannelReadsTest.scala]: Test Scope RawReads.rawChannelReads

File: raw/src/main/scala/de/welt/contentapi/raw/models/RawFormats.scala [hunk 3]
```diff
@@ -150,7 +150,34 @@ object RawReads {
...
+        config ← underlying.get("config").map(_.as[RawChannelConfiguration])
+        metadata ← underlying.get("metadata").map(_.as[RawMetadata])
+      } yield {
+        val maybeDeprecatedStages = underlying.get("stages").flatMap(_.asOpt[Seq[RawChannelStage]])
+        JsSuccess(
+          RawChannel(
+            id = id,
+            config = config,
+            stages = maybeDeprecatedStages,
+            stageConfiguration = Some(RawChannelStageConfiguration(stages = maybeDeprecatedStages)),
```
Comment (bobaaaaa) [inline raw/src/main/scala/de/welt/contentapi/raw/models/RawFormats.scala]: Fehlen hier nicht die "neuen" Werte?

**Comment by harryurban** [comment]
test this please

File: raw/src/main/scala/de/welt/contentapi/raw/models/RawFormats.scala (hunk 3 above)
Comment (harryurban) [inline raw/src/main/scala/de/welt/contentapi/raw/models/RawFormats.scala]: sehr richtig - behoben!

**Pull request merged**